### PR TITLE
feat(celestia-client): add utility function to access jsonrpc error

### PR DESCRIPTION
**Pull request by SuperFluffy**
This commit provides the `jsonrpc_call` utility method on `astria-celestia-jsonrpc-client::Error` that returns a jsonrpsee `Option<&ErrorObject<'_>>` only if the underlying is an actual jsonrpc error response.

To be used like so:
```rust
match self.client.blob_get_all(req).await {
    Ok(rsp) => rsp,
    Err(e) => if let Some(jsonrpc_call) = e.jsonrpc_call() {
        println!("jsonrpc returned error response: {}", jsonrpc_call.message());
    }
```

### CODE REVIEW

File: crates/astria-celestia-jsonrpc-client/src/lib.rs [hunk 1]
```diff
@@ -71,6 +72,17 @@ impl Error {
         self.rpc
     }
 
+    /// Convenience method to access the error returned by a JSONRPC.
+    ///
+    /// Returns `None` if the inner error is not related to the JSONRPC response.
+    #[must_use]
+    pub fn jsonrpc_call(&self) -> Option<&ErrorObject<'_>> {
```
[the remark this answers is not in the document]
Comment (joroshiba) [reply crates/astria-celestia-jsonrpc-client/src/lib.rs:79]: `jsonrpc_error_check`?

[the remark this answers is not in the document]
Comment (noot) [reply crates/astria-celestia-jsonrpc-client/src/lib.rs:79]: yeah i like `jsonrpc_response`

**Review by noot — APPROVED** [review]

File: crates/astria-celestia-jsonrpc-client/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (SuperFluffy) [reply crates/astria-celestia-jsonrpc-client/src/lib.rs]: Renamed to `jsonrpc_response` and also added a doc test to demonstrate usage.

**Review by joroshiba — APPROVED** [review]
If there are places we use this pattern could we do cleanup in this PR or have an immediate follow up?

Semantic sugar as dead code isn't likely to be a copied pattern.

**Comment by SuperFluffy** [comment]
@joroshiba 
> If there are places we use this pattern could we do cleanup in this PR or have an immediate follow up?
> 
> Semantic sugar as dead code isn't likely to be a copied pattern.
I linked #211 to use it

**Pull request merged**